authn/config/config.go

```diff
@@ -0,0 +1,54 @@
+package config
+
+import (
+	"strings"
+
+	"github.com/aserto-dev/go-utils/authn/apikey"
+	"github.com/aserto-dev/go-utils/authn/auth0"
+)
+
+type Config struct {
+	APIKeys apikey.Config `json:"api_keys"`
+	Auth0   auth0.Config  `json:"auth0"`
+	Options CallOptions   `json:"options"`
+}
+
+type CallOptions struct {
```

Comment:
Why do we need this? Why wasn't the old CallConfig map style enough?

Reply:
@gertd suggested we revise the old CallConfig and make it more compact without all the boilerplate. This model allows us to set default options and only override them for endpoints that are different.